feature/google/src/main/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -217,13 +217,20 @@ class BillingWrapper(
             presentedOfferingsByProductIdentifier[storeProduct.sku] = presentedOfferingIdentifier
         }
         executeRequestOnUIThread {
-            val params = createPurchaseParams(
-                storeProduct,
-                purchaseOption,
-                replaceSkuInfo,
-                appUserID
-            ) ?: return@executeRequestOnUIThread
-            launchBillingFlow(activity, params)
+            try {
+                val params = createPurchaseParams(
+                    storeProduct,
+                    purchaseOption,
+                    replaceSkuInfo,
+                    appUserID
+                )
+                launchBillingFlow(activity, params)
+            } catch (exception: InvalidProductException) {
```

Comment:
We aren't using exceptions much in our SDK, but I think it makes sense here. Lmk if you have any concerns over this though and I can refactor it! @vegaro

Reply:
I think the only concern would be if someone calls `createPurchaseParams` in the future in another part of the code and forgets to catch the exception. In that case, the exception will make it to the app right?

Reply:
Yeah... It's a valid concern... I think the best alternative is to move the validation to the caller or to a different function so we can handle it there... Will play with it a bit more. Thanks for the feedback!

Reply:
couldn't we mark createPurchaseParams with `throws` to enforce catching the exception?

Reply:
I believe in kotlin we can't add that to the signature. We can add an annotation but that only affects Java interoperability if I'm correct. Do lmk if you think we can do it though! 

Also, i think the latest version is cleaner considering we won't need to check for token nullability soon.